### ISSUE TITLE
How to add an additional lip shape?

### ISSUE THREAD

**Issue by Fossiel** [DanielSWolf/rhubarb-lip-sync#159]
Hi

I assume the project isn't actively being maintained anymore but I really hope you'll be able and willing to help me please.

I'm automating a simple rig for facial animation that will be driven by Rhubarb's XML output and I need to add an additional lip shape for the EE (like in bee) sound. (For example, having the new shape indicated by "Z" in the XML.) This would enable me to open the mouth wider when required, without messing up the consonants which require B.

Could you please tell me everything I would need to change in the source code in order to accomplish this?

I'm running Linux Mint and I've only once compiled a CMake project before. I recall it wasn't a joyride, given my limited knowledge of CMake and C++. Could you also please tell me the proper procedure and required libraries and compiler for compiling the project?

I know I'm asking for a lot but I would really appreciate it if you could assist me with this, please.

Thank you in advance

**Comment by DanielSWolf** [comment]
I can certainly give you some pointers. For adding mouth shapes, take a look at the discussion on #100.

Regarding building: You should just need CMake, a halfway-recent C/C++ compiler, and a not-too-ancient installation of Boost. Probably also a JDK for the extras. Then, follow these steps:

1. Create an empty directory `/build`
2. Move into this directory
3. Run `cmake ..` to configure CMake. Optionally, pass `-G ...` and/or `-D CMAKE_C_COMPILER=...` -- for inspiration, see `.github\workflows\ci.yml`
4. Build Rhubarb by running `cmake --build . --config Release`

**Comment by Fossiel** [comment]
Wow, that was fast. Thank you so, so much.

Before I contacted you, I had looked through the code and I saw that all the shape definitions appeared to be in the "animation" directory. I saw the matrix, which you had explained in the issue you linked, but couldn't make heads or tails of it. Thanks to your explanation, I should be able to add another row and column. It won't take long to go through the remaining files which contain a capital "B" to find all the spots where I need to add the additional shape.

Thank you very much.
